apps/admin/backend/src/server.ts

```diff
@@ -109,528 +163,279 @@ function buildApi(auth: DippedSmartCardAuthApi, workspace: Workspace) {
-    Admin.PutWriteInTranscriptionResponse,
-    Admin.PutWriteInTranscriptionRequest
-  >('/admin/write-ins/:writeInId/transcription', (request, response) => {
-    const { writeInId } = request.params;
-    const parseResult = safeParse(
-      Admin.PutWriteInTranscriptionRequestSchema,
-      request.body
-    );
+      if (addFileResult.ok().wasExistingFile) {
+        // log failure if the file was a duplicate
+        await logger.log(LogEventId.CvrLoaded, userRole, {
+          message:
+            'CVR file was not loaded as it is a duplicate of a previously loaded file.',
+          disposition: 'failure',
+          filename,
+          result: 'File not loaded, error shown to user.',
+        });
+      } else {
+        // log success otherwise
+        await logger.log(LogEventId.CvrLoaded, userRole, {
+          message: 'CVR file successfully loaded.',
+          disposition: 'success',
+          filename,
+          numberOfBallotsImported: addFileResult.ok().newlyAdded,
+          duplicateBallotsIgnored: addFileResult.ok().alreadyPresent,
+        });
+      }
 
-    if (parseResult.isErr()) {
-      response.status(400).json({
-        status: 'error',
-        errors: [
-          {
-            type: 'ValidationError',
-            message: parseResult.err().message,
-          },
-        ],
-      });
-      return;
-    }
+      return addFileResult;
+    },
 
-    store.transcribeWriteIn(writeInId, parseResult.ok().value);
-    response.json({ status: 'ok' });
-  });
+    clearCastVoteRecordFiles() {
+      const electionId = loadCurrentElectionIdOrThrow(workspace);
+      store.deleteCastVoteRecordFiles(electionId);
+      store.setElectionResultsOfficial(electionId, false);
+    },
 
-  deprecatedApiRouter.get<
-    { electionId: Id },
-    Admin.GetWriteInAdjudicationsResponse,
-    Admin.GetWriteInAdjudicationsRequest,
-    Admin.GetWriteInAdjudicationsQueryParams
-  >(
-    '/admin/elections/:electionId/write-in-adjudications',
-    (request, response) => {
-      const parseQueryResult = safeParse(
-        Admin.GetWriteInAdjudicationsQueryParamsSchema,
-        request.query
+    getCastVoteRecordFileMode() {
+      return store.getCurrentCvrFileModeForElection(
+        loadCurrentElectionIdOrThrow(workspace)
       );
+    },
 
-      if (parseQueryResult.isErr()) {
-        response.status(400).json({
-          status: 'error',
-          errors: [
-            {
-              type: 'ValidationError',
-              message: parseQueryResult.err().message,
-            },
-          ],
-        });
-        return;
-      }
-
-      const { electionId } = request.params;
-      const { contestId } = parseQueryResult.ok();
-
-      const writeInAdjudications = store.getWriteInAdjudicationRecords({
-        electionId,
+    getWriteIns({
+      contestId,
+      status,
+      limit,
+    }: {
+      contestId?: ContestId;
+      status?: Admin.WriteInAdjudicationStatus;
+      limit?: number;
+    } = {}) {
```

Comment:
Two conventions I thought might be nice when I was writing the first grout API:
- Always name the single parameter `input` (makes it a bit easier to read the signature, and also helps with type tooltips). Destructure later if needed
- Always write an explicit return type (not strictly needed, but since we have a lot of type inference happening inside grout, I thought it might help with clarity to readers)